src/sql/sqlops.proposed.d.ts

```diff
@@ -1365,4 +1365,620 @@ declare module 'sqlops' {
 		 */
 		export function openConnectionDialog(providers?: string[], initialConnectionProfile?: IConnectionProfile, connectionCompletionOptions?: IConnectionCompletionOptions): Thenable<connection.Connection>;
 	}
+
+	export namespace nb {
+		export function registerNotebookProvider(provider: NotebookProvider): vscode.Disposable;
+
+		export interface NotebookProvider {
+			handle: number;
```

Comment:
Our other providers don't expose a `handle` since it needs to be set by Azure Data Studio when the provider is registered and not by the extension, otherwise the extension wouldn't know how to get a unique handle. Or am I misunderstanding this?

Reply:
So... all the providers in the sqlops namespace define this (everything that extends from DataProvider). However I note that none of the vscode ones do (they use internally but don't register). 

I will remove the handle from here and look to work around it if needed. Maybe in the future we can look at removing from the other Providers too?